cvmfs/kvstore.h

```diff
@@ -107,22 +111,41 @@ class MemoryKvStore :SingleCopy {
     , max_entries_(cache_entries)
     , entries_(cache_entries, shash::Any(), lru::hasher_any,
         statistics, name)
+    , heap_(NULL)
     , counters_(statistics, name + ".lru") {
     int retval = pthread_rwlock_init(&rwlock_, NULL);
     assert(retval == 0);
-    if (alloc == kMallocArena) {
+    switch (alloc) {
+    case kMallocArena:
       malloc_arenas_.push_back(new MallocArena(kArenaSize));
+      break;
+    case kMallocHeap:
+      heap_ = new MallocHeap(alloc_size,
+          this->MakeCallback(&MemoryKvStore::OnBlockMove, this));
+      break;
+    default:
+      break;
     }
   }
 
   ~MemoryKvStore() {
```

Comment:
I think allocator and deallocator became large enough to put them in the .cc file